src/schemas/index.ts

```diff
@@ -0,0 +1,26 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+export enum SchemaVersion {
+	v1_0 = '',
+	v2_0 = '2',
+	v2_1 = '2.1',
+}
+
+export const DEFAULT_SCHEMA_VERSION = SchemaVersion.v2_1;
+
+const schemas: any = {};
+schemas[SchemaVersion.v1_0] = 'v1';
+schemas[SchemaVersion.v2_0] = 'v2.0';
+schemas[SchemaVersion.v2_1] = 'v2.1';
+
+function loadJson(p: string): any {
+	const buf = fs.readFileSync(p);
+	return JSON.parse(buf.toString('utf-8'));
+}
+
+export function getSchema(version: SchemaVersion): any {
+	const fileVersion = schemas[version];
+	const p = path.join(__dirname, `./config_schema_${fileVersion}.json`);
```

Comment:
Probably more "correct" to use `require.resolve` here.

Reply:
TS is slightly (but understandably) annoying with loading JSON -- it requires that `require` only loads modules (hence the loading via `fs` here). So, in that sense, I prefer treating the JSON files as bare resources and not tie them with the module-loading system.

Reply:
I meant `require.resolve(a)` instead of `path.join(__dirname, a)`, which just returns a string, but from the executing file's directory. I think they are equivalent for all intents and purposes, I've just always seen it done the other way.

e.g
```typescript
const p = require.resolve('./config_schema_${fileVersion}.json')
return loadJson(p);

Reply:
Cool, done.